src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/SourcelinkTests.cs

```diff
@@ -0,0 +1,134 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+public class SourcelinkTests : SmokeTests
+{
+    private static string SourcelinkRoot { get; } = Path.Combine(Directory.GetCurrentDirectory(), "sourcelink");
+
+    private string sourcelinkToolPath;
+    private ConcurrentBag<string> failedFiles = new ConcurrentBag<string>();
+
+    public SourcelinkTests(ITestOutputHelper outputHelper) : base(outputHelper) { }
+
+    /// <summary>
+    /// Verifies that all symbols have valid sourcelinks.
+    /// </summary>
+    [SkippableFact(Config.ExcludeSourcelinkEnv, skipOnTrueEnv: true)]
+    public void VerifySourcelinks()
+    {
+        if (Directory.Exists(SourcelinkRoot))
+        {
+            Directory.Delete(SourcelinkRoot, true);
+        }
+        Directory.CreateDirectory(SourcelinkRoot);
+
+        sourcelinkToolPath = GetSourcelinkToolPath();
+
+        ValidateAllFiles(ExtractPackages(GetAllSymbolsPackages()));
+
+        foreach (string file in failedFiles)
+        {
+            OutputHelper.WriteLine($"Failed sourcelink verification: {file}");
+        }
+
+        Assert.True(failedFiles.Count == 0);
+    }
+
+    /// <summary>
+    /// Gets the path to sourcelink binary.
+    /// Extracts the dotnet-sourcelink tool package from PSB arhive.
+    /// </summary>
+    /// <returns>Path to sourcelink tool binary.</returns>
+    private string GetSourcelinkToolPath()
+    {
+        string sourcelinkToolPackageNamePattern = "dotnet-sourcelink*nupkg";
+        string sourcelinkToolBinaryFilenamePattern = "dotnet-sourcelink.dll";
+
+        string sourcelinkRoot = Directory.CreateDirectory(Path.Combine(SourcelinkRoot, "sourcelink-tool")).FullName;
+        Utilities.ExtractTarball(Config.SourceBuiltArtifactsPath, sourcelinkRoot, sourcelinkToolPackageNamePattern);
+        string[] files = Directory.GetFiles(sourcelinkRoot, sourcelinkToolPackageNamePattern, SearchOption.AllDirectories);
+        Assert.True(files.Length > 0, "Did not find sourcelink tool package in PSB Artifacts archive");
```

Comment:
Perhaps this should verify there is only 1 file instead since that's what's expected. This applies to several locations in the code where it's just expecting one item (e.g. `files[0]`).

Reply:
Thanks - I'll update the checks.

Reply:
Fixed with https://github.com/dotnet/installer/pull/17309/commits/9ff1e023cececde9d7f33c7e5d4d9e093347c249